libquickevent/libquickeventcore/src/resultstatus.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "quickeventcoreglobal.h"
+#include <qf/core/sql/query.h>
+#include <qf/core/utils/treetable.h>
+
+namespace quickevent {
+namespace core {
+
+class QUICKEVENTCORE_DECL_EXPORT ResultStatus
+{
+public:
+	ResultStatus() = default;
+	ResultStatus(qf::core::sql::Query &q);
+	ResultStatus(const qf::core::utils::TreeTableRow &ttr);
+	// set all variables from query (table runs)
+	void fillFromQuery(qf::core::sql::Query &q);
+	void fillFromTreeTableRow(const qf::core::utils::TreeTableRow &ttr);
+
+	bool isOk() const;
+	int getOGTime(int time) const;
+
+	QString statusXml() const;
+	QString statusEmmaTxt() const;
+	QString statusResultExport() const;
+	QString statusText() const;
+
+	bool disqualified() const { return m_disqualified; }
+	bool disqualifiedByOrganizer() const { return m_disqualifiedByOrganizer; }
+	bool notCompeting() const { return m_notCompeting; }
+	bool missingPunch() const { return m_missingPunch; }
+	bool didNotStart() const { return m_didNotStart; }
+	bool didNotFinish() const { return m_didNotFinish; }
+	bool overTime() const { return m_overTime; }
+
+	static QString dbRunsColumnList();
```

Comment:
`static QStringList runsTableColumns();`

Reply:
Prejmenuju.

Reply:
Muzu predelat seznam promennych pro potreby SQL do listu.
Ale bez prevodu do stringu jsem nenasel zpusob jak to do QueryBuilderu dostat.
Nejaka napoveda, co jsem prehledl ? Nebo to opravdu je potreba udelat dvoufazove ?

Reply:
myslel jsem `QStringList::join()`, je to sice operace navic, ale return type QStringList rika, ze se jedna o list nazvu, nemusi se resit oddelovac, ktery je v SQL carka, ale pak by v nazvu funkce zase nekde melo byt Sql.

Reply:
Tak tuhle funkci jsem minul. Chapu. Opravim.